packages/driver/src/cypress/command_queue.ts

```diff
@@ -254,20 +262,18 @@ export class CommandQueue extends Queue<Command> {
   // @ts-ignore
   run () {
     const next = () => {
+      // start at 0 index if one is not already set
+      let index = this.state('index') || this.state('index', 0)
+
       // bail if we've been told to abort in case
       // an old command continues to run after
       if (this.stopped) {
         return
       }
 
-      // start at 0 index if we dont have one
```

Comment:
Why did this need to move out of curiosity?

Reply:
I don't think it's necessary. Probably an interstitial change that didn't end up being needed. Reverted in https://github.com/cypress-io/cypress/pull/21137/commits/5efb24a0c3a6c51be146009750e97af1138c0ee9